### PR TITLE
[Anchor -378] Retry at most 3 times for connection error

### PR DESCRIPTION
**Pull request by JiahuiWho**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Change to retry at most 3 times for connection error

### Why

Please refer to comments and discussions here: https://github.com/stellar/java-stellar-anchor-sdk/pull/1044#issuecomment-1673571967

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java [hunk 1]
```diff
@@ -246,20 +246,21 @@ void handleEvent(AnchorEvent event) {
...
-            currentThread().interrupt();
+          // Retry for connection error
+          if (++networkErrorRetryAttempts <= NETWORK_MAX_RETRIES) {
+            try {
+              getNetworkBackoffTimer().backoff();
+            } catch (InterruptedException e) {
+              // The thread is interrupted, so we need to stop the processor. This will
+              // break the while loop.
+              currentThread().interrupt();
+            }
```
Comment (JakeUrban) [inline platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java:260]: Why are we catching an `InterruptedException` here for the network backoff timer but not for the http status backoff timer?

**Review by JakeUrban — APPROVED** [review]
LGTM with two questions

Right now we're allowing up to 5 requests per event (2 http errors, 2 connection errors, and one success or 1 additional http or connection error). Do we want to limit it to 3 requests total?

File: platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java (hunk 1 above)
Comment (JiahuiWho) [reply platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java:260]: Will catch for HttpErrorBackoffTimer as well

**Comment by JiahuiWho** [comment]
I was debating between if we want to have separate counter for each exception or just combine them in one. To me it's not likely that we will have that kind of mixed exception as you mentioned, and by isolating them we can easily pinpoint the specific issues causing certain exceptions, which also benefits diagnosing and addressing the root causes. 

But I don't have a strong preference on this, we can certainly change it to at most 3 in total for efficiency.

**Pull request merged**